src/AutoMapper/Mappers/FromStringDictionaryMapper.cs

```diff
@@ -33,16 +34,24 @@ public class FromStringDictionaryMapper : IObjectMapper
         private static TDestination Map<TDestination>(StringDictionary source, TDestination destination, ResolutionContext context, ProfileMap profileMap)
         {
             var destTypeDetails = profileMap.CreateTypeDetails(typeof(TDestination));
-            var members = from name in source.Keys
-                          join member in destTypeDetails.PublicWriteAccessors on name equals member.Name
-                          select member;
+
+            var memberMatches = from member in destTypeDetails.PublicWriteAccessors
+                                join key in source.Keys on member.Name equals key.Trim() into matchingKeys
+                                where matchingKeys.Any()
+                                select new { member, sourceNames = matchingKeys };
+
             object boxedDestination = destination;
-            foreach (var member in members)
+            foreach (var match in memberMatches)
             {
-                var value = context.MapMember(member, source[member.Name], boxedDestination);
-                member.SetMemberValue(boxedDestination, value);
+                if (match.sourceNames.Count() > 1)
+                {
+                    throw new InvalidOperationException($"Multiple matching keys were found in the source dictionary for {match.member.Name}.");
```

Comment:
Throw `AutoMapperMappingException` instead. And be sure to specify the types being mapped. We want to give as much details as possible.

Reply:
As far as I can tell, this exception gets wrapped as an InnerException in an `AutoMapperMappingException` - which has some type info.

If I threw an `AutoMapperMappingException` instead, wouldn't it just get wrapped as well? (resulting in an `AutoMapperMappingException` containing an `AutoMapperMappingException`)

Reply:
OK, but the more important point is what's in that exception. I'm guessing it does not contain the type pair we're mapping, but if it does, then you're home free. In your test, check the type pair in the exception.